server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -1504,17 +1504,15 @@ else if (!poolQuantities.keySet().equals(entitlements.keySet())) {
         // we might have changed the bonus pool quantities, lets find out.
         handler.handleBonusPools(poolQuantities, entitlements);
 
-        JobDetail detail = ConsumerComplianceJob.scheduleWithForceUpdate(consumer);
-        detail.getJobDataMap().put(PinsetterJobListener.PRINCIPAL_KEY, new SystemPrincipal());
-
-        log.info("Triggering ConsumerComplianceJob: {} for consumer: {}", detail.getKey(),
-            consumer.getUuid());
 
-        try {
-            pinsetterKernel.scheduleSingleJob(detail);
-        }
-        catch (PinsetterException e) {
-            log.error("ConsumerComplianceJob schedule failed", e.getMessage());
+        /*
+         * If the consumer is not a distributor, check consumer's new compliance
+         * status and save. the getStatus call does that internally, so we only
+         * need to check for the update.
+         */
+        complianceRules.getStatus(consumer, null, false, false);
+        if (!isDistributor) {
+            consumerCurator.update(consumer);
```

Comment:
@nguyenfilip this PR mostly reverts code to its prior state, so this was pre-existing behaviour.
I *think* the intent was that in getStatus, the update is only if complianceChanged or entStatusChanged, while this way its unconditionally changed.

that being said, I dont see what other changes could be made in this method itself, but this is a common method used in a lot of places so I found it safer to keep it this way.
thoughts?